### ISSUE TITLE
HiDrive Strato cloud storage support

### ISSUE THREAD

**Issue by highend** [rclone/rclone#1069]
Hello,

Is there any chance that this provider can be supported as well? It has a well documented api available here: https://dev.strato.com/hidrive/documentation

It's homepage is: https://dev.strato.com/hidrive/documentation
And a free account is available at: https://www.free-hidrive.com/

Grazie mille!

**Comment by ncw** [comment]
Any volunteers?

**Comment by Frank-Buss** [comment]
I tried it with webdav and the HiDrive service from ionos.com, because webdav is integrated in the cheaper accounts, for smb or rsync you have to pay extra. It worked like this:
```
rclone config create hidrive webdav url https://webdav.hidrive.ionos.com vendor other user myusername pass mypassword
rclone copy hidrive:users/myusername/test.txt .
```
With "myusername" your username as showed in "Settings" of the HiDrive web interface. This copied the file "test.txt" from HiDrive to my local system. Works with directories and for uploads as well.

**Comment by ncw** [comment]
Thanks for testing @Frank-Buss - that looks like a pretty standard WebDAV setup.

Is this worth writing up in the docs? How popular is HiDrive?

**Comment by Frank-Buss** [comment]
I don't know how popular it is. But searching for HiDrive in Google results in 490,000 pages, and it is pretty cheap with EUR 1 per month for the first 12 months for 100 GB. And apparently there is no additional transfer costs, like you have to pay for other services, e.g. Google Cloud storage. Might be useful to add it to the documentation. The "users/myusername" part and the URL was not obvious and I needed some time to find it out.

**Comment by ncw** [comment]
Do you want to have a go at adding it to the docs @Frank-Buss ? The file in question is this one

https://github.com/rclone/rclone/blob/master/docs/content/webdav.md

**Comment by yapsr** [comment]
Strato HiDrive is currently on sale for 1TB for 1€/month.

**Comment by m-ueberall** [comment]
> Strato HiDrive is currently on sale: 1TB for 1€ for the first year.

Not sure whether this is constantly the case, but atm they still/again advertise the same rate ("Easter Savings"), which also includes smb/rsync among others. Apart from the domain name (`strato.com` instead of `ionos.com`), everything works using the same setup instructions used by @Frank-Buss above.

**Comment by awehrfritz** [comment]
Did someone figure out whether HiDrive by Strato is actually the same as HiDrive by IONOS? They are in fact still two different companies and the products are marketed separately, though it is worth to note that both providers are owned nowadays by the same company, i.e. United Internet AG (https://www.united-internet.de/en/).

Anyway, focusing on HiDrive by Strato, while they offer WebDAV access for some of their products that come with an extended "protocol package", the default way of accessing HiDrive is via a REST web API documented here: https://developer.hidrive.com/http-api-reference/
In fact, HiDrive accounts with the protocol package can easily be accessed via FTP, SFTP, rsync or even SMB, all preferable options over WebDAV.

In summary, I believe there is still value in supporting the HiDrive REST API as it offers access without any add-on packages. (Personally, my HiDrive account comes with the protocol package and I use SFTP/sshfs to access my data from my desktop; SSH/SFTP access was actually the primary reason why I signed up for it in the first place.)
